src/webots/nodes/WbSphere.cpp

```diff
@@ -116,13 +107,17 @@ bool WbSphere::areSizeFieldsVisibleAndNotRegenerator() const {
 
 void WbSphere::exportNodeFields(WbVrmlWriter &writer) const {
   WbGeometry::exportNodeFields(writer);
-  if (writer.isX3d())
-    writer << " subdivision=\'" << 8 * mSubdivision->value() << ',' << 8 * mSubdivision->value() << "\'";
+  if (writer.isX3d()) {
+    writer << " subdivision=\'" << mSubdivision->value() << ',' << mSubdivision->value() << "\'";
+    writer << " ico=\'" << (mIco->value() ? "true" : "false") << "\'";
+  }
 }
 
 bool WbSphere::sanitizeFields() {
-  if (WbFieldChecker::checkIntInRangeWithIncludedBounds(this, mSubdivision, 1, 6, 1))
-    return false;
```

Comment:
I don't fully understand why it is needed to return false (and skip WREN mesh creation) if a field has not a correct value and it is reset to a default value.
The problem if that if the mesh is not created, the next time you edit/select or interact with the node Webots crashes.

Reply:
This mechanism is indeed questionable. It is present in many geometric primitives. We should either keep it everywhere or remove it everywhere. But if we do so, it should be in done in a different PR. If you believe it is wrong, you should open an issue (question) about it.

Reply:
This mechanism is difficult to understand, but useful: this is because `WbFieldChecker` functions [actually change the field value](https://github.com/omichel/webots/blob/master/src/webots/nodes/utils/WbFieldChecker.cpp#L255) if required, and so triggers itself the mesh recreation (if it's not the case, there is maybe a signal-slot issue). Therefore returning is important.

Reply:
I will inspect the issue once completed this PR and open issue if it is required.
But it would help to better understand why the WREN creation is skipped.

Reply:
And also, the method name is probably not well chosen. Instead of `checkIntInRangeWithIncludedBounds`, it should probably be named something like `clampIntInRangeWithIncludedBounds`.

Reply:
The `WbFieldChecker` always changes the field value for every node, but only for the geometries we have this more complex workflow that causes crashes with WREN.
Anyway I will check this later, and if it possible to simplify it and make it more robust I think it would be better.